lte/gateway/c/session_manager/ShardTracker.cpp

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright 2020 The Magma Authors.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <string>
+#include <utility>
+#include <vector>
+#include <algorithm>
+#include <map>
+
+#include "ShardTracker.h"
+#define MAX_SHARD_SIZE 100
+
+namespace magma {
+
+ShardTracker::ShardTracker() {
+  number_of_shards = 0;
+}
+
+int ShardTracker::add_ue() {
```

Comment:
This function seems to be doing a lot. Can we add some comments to make it easier to read?